### PR TITLE
Fix NumberFormatException when loadng persisted datasource pool config

### LINKED ISSUE #27927: There are "null" strings in the persisted connection pool parameter
### Which version of ShardingSphere did you use?
master 73a3e3d04edf9c8dfc88fb6710e483d8433f6a9b

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Expected behavior
null values are not persisted

### Actual behavior
Null values are persisted, causing restart exceptions
#### yaml
```
databaseName: encrypt_db

dataSources:
  unique_ds:
    url: jdbc:mysql://localhost:3306/demo_ds?serverTimezone=UTC&useSSL=false&useUnicode=true&characterEncoding=UTF-8
    username: root
    password: 123456
```

#### persisted value
```yaml
password: '123456'
minPoolSize: null
connectionTimeoutMilliseconds: null
maxLifetimeMilliseconds: null
dataSourceClassName: com.zaxxer.hikari.HikariDataSource
readOnly: null
idleTimeoutMilliseconds: null
maxPoolSize: null
url: jdbc:mysql://localhost:3306/demo_ds?serverTimezone=UTC&useSSL=false&useUnicode=true&characterEncoding=UTF-8
username: root

```

#### log
```
Exception in thread "main" java.lang.NumberFormatException: For input string: "null"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Long.parseLong(Long.java:711)
	at java.base/java.lang.Long.valueOf(Long.java:1163)
	at org.apache.shardingsphere.infra.datasource.props.DataSourcePropertiesCreator.getPoolConfiguration(DataSourcePropertiesCreator.java:147)
	at org.apache.shardingsphere.infra.datasource.props.DataSourcePropertiesCreator.createConfiguration(DataSourcePropertiesCreator.java:135)
	at org.apache.shardingsphere.metadata.persist.NewMetaDataPersistService.lambda$getEffectiveDataSources$2(NewMetaDataPersistService.java:151)
	at java.base/java.util.stream.Collectors.lambda$toMap$68(Collectors.java:1674)
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.apache.shardingsphere.metadata.persist.NewMetaDataPersistService.getEffectiveDataSources(NewMetaDataPersistService.java:150)
	at org.apache.shardingsphere.mode.metadata.NewMetaDataContextsFactory.createEffectiveDatabaseConfiguration(NewMetaDataContextsFactory.java:121)
	at org.apache.shardingsphere.mode.metadata.NewMetaDataContextsFactory.lambda$createEffectiveDatabaseConfigurations$1(NewMetaDataContextsFactory.java:116)
	at java.base/java.util.stream.Collectors.lambda$toMap$68(Collectors.java:1674)
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.apache.shardingsphere.mode.metadata.NewMetaDataContextsFactory.createEffectiveDatabaseConfigurations(NewMetaDataContextsFactory.java:115)
```

### PR DESCRIPTION
**Pull request by zhangfengcdt**
Fixes #27927.

Changes proposed in this pull request:
  - handling correctly when parsing non-numeric values

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [X] I have self-reviewed the commit code.
- [X] I have (or in comment I request) added corresponding labels for the pull request.
- [X] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [X] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by zhaojinchao95** [comment]
@RaigorJiang Hi, can you review it?

File: infra/datasource/core/src/test/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesCreatorTest.java [hunk 1]
```diff
@@ -81,4 +102,26 @@ private Map<String, Object> createProperties() {
         result.put("maximumPoolSize", "-1");
         return result;
     }
+    
+    private Map<String, Object> createStandardProperties() {
+        Map<String, Object> result = new LinkedHashMap<>(6, 1F);
+        // This is to test the case where the property type is mismatched.
```
Comment (RaigorJiang) [inline infra/datasource/core/src/test/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesCreatorTest.java:108]: Hi @zhangfengcdt 
Thanks for your PR!
According to [Code of Conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/), comments can only be used for Javadoc, todo and fixme, so please remove this line of comment

Comment (zhangfengcdt) [reply infra/datasource/core/src/test/java/org/apache/shardingsphere/infra/datasource/props/DataSourcePropertiesCreatorTest.java]: Sure, comment removed.

**Review by RaigorJiang — APPROVED** [review]

**Comment by RaigorJiang** [comment]
Merged, thank you! @zhangfengcdt

**Comment by zhangfengcdt** [comment]
> Merged, thank you! @zhangfengcdt

Thank you, @RaigorJiang for the review!

**Pull request merged**